microprofile-health/src/test/java/org/jboss/eap/qe/microprofile/health/manual/delay/NotReadyHealthCheckTest.java

```diff
@@ -0,0 +1,233 @@
+ */
+@Category(ManualTests.class)
+@RunWith(Arquillian.class)
+public class NotReadyHealthCheckTest {
+
+    @ArquillianResource
+    ContainerController controller;
+
+    private ExecutorService executorService;
+
+    /**
+     * Builds a deployment that contains just a readiness health check which has a constructor that simulates
+     * a delay in returning the accurate readiness status (e.g.: like one testing a database connection availability
+     * could do)
+     *
+     * @return A {@link File} instance that refers to the ZIP archive that will be copied to Wildfly {@code deployments}
+     *         directory
+     */
+    public static File delayedDownReadinessCheckDeployment() {
+        File deployment;
+        try {
+            Path tempDirectory = Files.createTempDirectory(null);
+            deployment = new File(tempDirectory.toFile(), NotReadyHealthCheckTest.class.getSimpleName() + ".war");
+        } catch (IOException e) {
+            e.printStackTrace();
+            deployment = new File(NotReadyHealthCheckTest.class.getSimpleName() + ".war");
+        }
+
+        ShrinkWrap.create(WebArchive.class, NotReadyHealthCheckTest.class.getSimpleName() + ".war")
+                .addClasses(DelayedReadinessHealthCheck.class)
+                .addAsWebInfResource(EmptyAsset.INSTANCE, "beans.xml")
+                .as(ZipExporter.class)
+                .exportTo(deployment, true);
+
+        return deployment;
+    }
+
+    /**
+     * Builds a deployment that contains just a liveness - thus <b>no readiness</b> - health check which has a
+     * constructor that simulates a delay in returning the accurate liveness status (e.g.: like one acting as an AMQ
+     * backup could do)
+     *
+     * @return A {@link File} instance that refers to the ZIP archive that will be copied to Wildfly {@code deployments}
+     *         directory
+     */
+    public static File unexpectedReadinessChecksDeployment() {
+        File deployment;
+        try {
+            Path tempDirectory = Files.createTempDirectory(null);
+            deployment = new File(tempDirectory.toFile(), DelayedLivenessHealthCheck.class.getSimpleName() + ".war");
+        } catch (IOException e) {
+            e.printStackTrace();
+            deployment = new File(DelayedLivenessHealthCheck.class.getSimpleName() + ".war");
+        }
+
+        ShrinkWrap.create(WebArchive.class, DelayedLivenessHealthCheck.class.getSimpleName() + ".war")
+                .addClasses(DelayedLivenessHealthCheck.class)
+                .addAsWebInfResource(EmptyAsset.INSTANCE, "beans.xml")
+                .as(ZipExporter.class)
+                .exportTo(deployment, true);
+
+        return deployment;
+    }
+
+    /**
+     * Unmanaged empty deployment to state that Arquillian shouldn't care about deployments at all
```

Comment:
Is this method necessary? Can be removed?

Reply:
Thanks, removing it.